tests/org.eclipse.xtend.core.tests/src/org/eclipse/xtend/core/tests/validation/FeatureCallValidationTest.java

```diff
@@ -255,6 +256,53 @@ public void testBug400989() throws Exception {
 		assertEquals(1, issue.getLineNumber().intValue());
 	}
 	
+	@Test
+	public void testBug334499_01() throws Exception {
+		XtendClass clazz = clazz("class C { def void m() { 'str'.foo('str') }}");
+		helper.assertError(clazz, XbasePackage.Literals.XMEMBER_FEATURE_CALL, Diagnostic.LINKING_DIAGNOSTIC, "foo(String) is", "for the type String");
+	}
+	
+	@Test
+	public void testBug334499_02() throws Exception {
+		XtendClass clazz = clazz("class C { def void m() { this.foo('str',5) }}");
+		helper.assertError(clazz, XbasePackage.Literals.XMEMBER_FEATURE_CALL, Diagnostic.LINKING_DIAGNOSTIC, "foo(String, int) is", "for the type C");
+	}
+	
+	@Test
+	public void testBug334499_03() throws Exception {
+		XtendClass clazz = clazz("class C { def void m() { foo('',5) }}");
+		helper.assertError(clazz, XbasePackage.Literals.XFEATURE_CALL,Diagnostic.LINKING_DIAGNOSTIC, "foo(String, int) is");
+	}
+	
+	@Test
+	public void testBug334499_04() throws Exception {
+		XtendClass clazz = clazz("class C { def void m() { val a = #['str'] a.map[it.foo(5)] }}");
+		helper.assertError(clazz, XbasePackage.Literals.XMEMBER_FEATURE_CALL, Diagnostic.LINKING_DIAGNOSTIC, "foo(int) is", "for the type String");
+	}
+	@Test
+	public void testBug334499_05() throws Exception {
+		XtendClass clazz = clazz("class C { def void m() { this.foo(this) }}");
+		helper.assertError(clazz, XbasePackage.Literals.XMEMBER_FEATURE_CALL, Diagnostic.LINKING_DIAGNOSTIC, "foo(C) is", "for the type C");
+	}
+	
+	@Test
+	public void testBug334499_06() throws Exception {
+		XtendClass clazz = clazz("class C { def m() { new C(){ override m(){ foo(1)} }}}");
+		helper.assertError(clazz, XbasePackage.Literals.XFEATURE_CALL, Diagnostic.LINKING_DIAGNOSTIC, "foo(int) is");
```

Comment:
Could you please add a test for an unresolved call in an anonymous class with a this. receiver?